server/core/src/main/scala/org/infinispan/server/core/transport/ExtendedByteBuf.scala

```diff
@@ -25,6 +29,11 @@ object ExtendedByteBuf {
       }
    }
 
+   def readOptRangedBytes(bf: ByteBuf): Option[Array[Byte]] = {
+      val length = readUnsignedInt(bf)
+      if (length < 0) None else Some(readRangedBytes(bf, length))
```

Comment:
Those ```read/writeUnsigned``` delegate to ```read/WrintVInt```, and they work nice with negative numbers although docs says otheriwse. @galderz any thoughts on that?